Sources/TSCBasic/Lock.swift

```diff
@@ -178,13 +178,20 @@ public final class FileLock {
             throw FileSystemError(.notDirectory, lockFilesDirectory)
         }
         // use the parent path to generate unique filename in temp
-        var lockFileName = (resolveSymlinks(fileToLock.parentDirectory).appending(component: fileToLock.basename)).components.joined(separator: "_") + ".lock"
+        var lockFileName = (resolveSymlinks(fileToLock.parentDirectory)
+                                .appending(component: fileToLock.basename))
+                                .components.joined(separator: "_")
+                                .replacingOccurrences(of: ":", with: "_") + ".lock"
+#if os(Windows)
+        var lockFileUTF8 = lockFilesDirectory.appending(component: lockFileName).pathString.utf8.suffix(Int(MAX_PATH))
```

Comment:
This is supposed to be just the filename, so `lockFilesDirectory` gets prepended again below.

Reply:
This means you'll end up with `lockFilesDirectory` twice.

Looking at this more, I think the part about `AbsolutePath.root` was try avoid pasting something that starts with `/` as a subpath.  I kept that part unmodified but it seems odd, since the file name should never have that in the first place.

My view is that we should just compute a SHA1 of the whole path, whatever it is, and then use that as the filename.  But it's true that it would be less readable.

Reply:
Oh I see:  With non-Windows file paths, the `components` returns a `/` as the first path component.  So this is just stripping that out.

Taking a step back, I think what would make sense here would be to take just the components leading from the root of the path (whether that's `/` or `C:\` or whatever) and just using those as the basis for the filename.  All this is trying to do is to come up with a unique file name.

Reply:
Is there an API to return just the path and not the root?

Reply:
That would be better, however, I _think_ that this new version should address the path issue.

Reply:
> Is there an API to return just the path and not the root?

I don't believe that there is such an API in the TSC Path API today.